File: 274-h-index/h-index.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function (citations) {
    let max = Math.max(...citations)
    let count = Array(max + 1).fill(0)
    for (let i = 0; i < citations.length; i++) {
        count[citations[i]]++
    }
    let newArr = []
    for (let i = 0; i < count.length; i++) {
        while (count[i] > 0) {
            newArr.push(i)
            count[i]--
        }
    }
    let ans = 0
    for (let i = 0; i < newArr.length; i++) {
        if (newArr.length - i <= newArr[i]) {
            return newArr.length - i
        }
    }
    return 0
};

File: 274-h-index/h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        max_citations = max(citations)
        count = [0] * (max_citations + 1)

        for citation in citations:
            count[citation] += 1

        new_arr = []
        for i in range(len(count)):
            while count[i] > 0:
                new_arr.append(i)
                count[i] -= 1

        ans = 0
        for i in range(len(new_arr)):
            if len(new_arr) - i <= new_arr[i]:
                ans = len(new_arr) - i
                break

        return ans